cartridges/rvw_dev_console/cartridge/client/default/js/vue/components/App.vue

```diff
@@ -499,40 +523,99 @@ export default {
                         message: `<i class="fa fa-check-square"></i>&nbsp; Deleted <strong>${fileName}</strong>`,
                         duration: 3000,
                         dismissible: true,
-                        position: 'top-right'
+                        position: 'top'
                     });
                 } else {
                     this.$toast.open({
                         type: 'error',
                         message: `<i class="fa fa-exclamation-triangle"></i>&nbsp; Unable to Delete <strong>${fileName}</strong>`,
                         duration: 5000,
                         dismissible: true,
-                        position: 'top-right'
+                        position: 'top'
                     });
                 }
             }
         },
         editorDidMount(editor) {
+            var self = this;
             this.editor = editor;
 
             // Editor tries to call this on Theme Change, even though it's not needed in Vue
             if (typeof this.editor.setTheme === 'undefined') {
                 this.editor.setTheme = function(){};
             }
 
+            // Enable Key Bindings
+            this.editor.condition = {
+                canSave: editor.createContextKey('canSave', false),
+                canClear: editor.createContextKey('canClear', false),
+                canRun: editor.createContextKey('canRun', true),
+            };
+
+            // Add Save Command to Editor
+            editor.addAction({
+                id: 'dev-console-save',
+                label: 'Save',
+                keybindings: [
+                    monaco.KeyMod.chord(monaco.KeyMod.CtrlCmd | monaco.KeyCode.KEY_S)
+                ],
+                precondition: 'canSave',
+                keybindingContext: 'canSave',
+                contextMenuGroupId: 'dev_console',
+                contextMenuOrder: 1,
+                run: function() {
+                    self.saveFile();
+                    return null;
+                }
```

Comment:
What does this function do? Is this the callback for the keybind action?

Reply:
Correct.  Turns out the Editor had native keybinding actions that allowed you to trap them in the editor.  It also allowed us to have them show up as commands in the command pallet.  Just lost scope, hence the `self`, but otherwise ya, this is just the command to run if the editor triggers the key binding ( using the conditions that were defined ... like you cannot trigger a results output if there is no code in the editor )